### PR TITLE
Symfony5

### PR DESCRIPTION
**Pull request by rugbymauri**
updated for symfony5,

- removed deprecations

### CODE REVIEW

**Comment by lukasluecke** [comment]
@rugbymauri Please merge feature-branches (which this qualifies as) into `develop`, and then create a release from there (GitFlow).

You can just edit the PR and change the base branch:
Bildschirmfoto 2019-12-20 um 10 24 49

File: composer.json [hunk 1]
```diff
@@ -20,11 +20,11 @@
...
-        "php":                      ">=5.6",
-        "symfony/framework-bundle": "~3.0|~4.0",
-        "symfony/http-kernel":      "~3.0|~4.0",
-        "symfony/validator":        "~3.0|~4.0",
-        "whatwedo/core-bundle":     "^0.3"
+        "php":                      ">=7.2",
+        "symfony/framework-bundle": "~4.0|~5.0",
+        "symfony/http-kernel":      "~4.0|~5.0",
+        "symfony/validator":        "~4.0|~5.0",
+        "whatwedo/core-bundle":     "dev-symfony5"
```
Comment (lukasluecke) [inline composer.json]: This should be set to a proper version before merging, after the respective CoreBundle release is finished.

File: Filter/Type/AbstractAjaxAssociationFilterType.php [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+
+namespace whatwedo\TableBundle\Filter\Type;
+use Doctrine\Bundle\DoctrineBundle\Registry;
+use Doctrine\ORM\QueryBuilder;
+
+/**
+ * Class AjaxManyToManyFilterType
```
Comment (lukasluecke) [inline Filter/Type/AbstractAjaxAssociationFilterType.php]: I'd just remove these comments, as they don't add any value and are wrong more often than not (as is the case right now).

File: Filter/Type/AbstractAjaxAssociationFilterType.php [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+     */
+    protected $doctrine;
+
+    /**
+     * @var string $targetClass
+     */
+    protected $targetClass;
+
+    /**
+     * AjaxManyToManyFilterType constructor.
```
Comment (lukasluecke) [inline Filter/Type/AbstractAjaxAssociationFilterType.php]: I'd just remove these comments, as they don't add any value and are wrong more often than not (as is the case right now).

File: Filter/Type/AbstractAjaxAssociationFilterType.php [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+     * @param $column
+     * @param $targetClass
+     * @param $doctrine
+     * @param array $joins
```
Comment (lukasluecke) [inline Filter/Type/AbstractAjaxAssociationFilterType.php]: Could we maybe add better typehints here?

**Comment by rugbymauri** [comment]
@lukasluecke btw. i'm working actually to get a better Insights Score, lot of issues and refactoring

File: Table/DoctrineHierarchialTable.php [hunk 5]
```diff
@@ -1,142 +0,0 @@
...
-use Twig\Environment;
-use whatwedo\CoreBundle\Manager\FormatterManager;
-use whatwedo\TableBundle\Extension\ExtensionInterface;
-use whatwedo\TableBundle\Model\SimpleTableData;
-
-/**
- * Class DoctrineTable
- * @package whatwedo\TableBundle\Table
- */
-class DoctrineHierarchialTable extends Table
```
Comment (lukasluecke) [inline Table/DoctrineHierarchialTable.php]: What does this have to do with Symfony5, and why was this removed at all?

File: Resources/config/services.yml [hunk 6]
```diff
@@ -4,6 +4,9 @@ services:
         autoconfigure: true
         public: false
 
+        bind:
+            $kernelBundles: '%kernel.bundles%'
+
```
Comment (lukasluecke) [inline Resources/config/services.yml]: Do these parameters get exposed beyond the bundle boundaries, and can they potentially conflict with user defined binds? If so, we should rather pass them to the event listeners directly - if not, then this is fine 🙂

File: Table/DoctrineHierarchialTable.php (hunk 5 above)
Comment (rugbymauri) [reply Table/DoctrineHierarchialTable.php]: Class does not belong to the Bundle, it s project-Specific behavior

File: EventListener/FilterEventListener.php [hunk 7]
```diff
@@ -94,7 +94,6 @@ private function addQueryBuilderFilter()
 
                 $filter = $filterExtension->getFilters()[$column];
 
-                // TODO: automatically join (split field on '.')
```
Comment (lukasluecke) [inline EventListener/FilterEventListener.php]: @rugbymauri This should not be removed, seeing as it was not implemented yet. A `TODO` is not necessarily a bad thing, but should serve as a pointer for possible future enhancements.
